src/script/components/VerificationBadge/VerificationBadges.tsx

```diff
@@ -76,6 +81,41 @@ const useConversationVerificationState = (conversation: Conversation) => {
   return {MLS: mlsState, proteus: proteusVerificationState};
 };
 
+const useMLSStatus = (identity?: WireIdentity, userEntity?: User) => {
+  const [MLSStatus, setMLSStatus] = useState<MLSStatuses | undefined>(undefined);
+  const [user, setUser] = useState<User | undefined>(userEntity);
+  const {current: userState} = useRef(container.resolve(UserState));
+
+  useEffect(() => {
+    if (!identity) {
+      return;
+    }
+
+    if (!user) {
+      void (async () => {
+        const user = await waitFor(() =>
+          userState
+            .users()
+            .find(user => stringifyQualifiedId(user.qualifiedId) === stringifyQualifiedId(identity.qualifiedUserId)),
+        );
+        setUser(user);
+      })();
+    }
```

Comment:
nah, this doesn't belong here, the caller should be the one retrieving the `user`
Let's keep this component as stupid as possible

Reply:
sure. can do. 
I don't see the harm since everything that the logic needs is in one place.
But as stupid as it needs is a good way to see things as well.